src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs

```diff
@@ -129,11 +129,12 @@ private void ValidateRedirectUri(Uri redirectUri)
             brokerRequest.Add(BrokerParameter.ClientId, authenticationRequestParameters.AppConfig.ClientId);
             brokerRequest.Add(BrokerParameter.CorrelationId, authenticationRequestParameters.RequestContext.CorrelationId.ToString());
             brokerRequest.Add(BrokerParameter.ClientVersion, MsalIdHelper.GetMsalVersion());
+
             var realEnrollmentId = IntuneEnrollmentIdHelper.GetRawEnrollmentId();
-            if (!string.IsNullOrEmpty(realEnrollmentId))
-            {
-                brokerRequest.Add(BrokerParameter.IntuneEnrollmentIds, realEnrollmentId);
-            }
+            brokerRequest.Add(BrokerParameter.IntuneEnrollmentIds, realEnrollmentId ?? string.Empty);
```

Comment:
In Intune scenario, in which step of the flow would these properties be empty? Is it on the very first request when the device is unregistered? So broker will return an exception?
I assume in non-Intune scenario broker will just ignore these properties?

Reply:
This is a great question and the answer should be added as a comment in the code.